src/psyclone/psyir/symbols/symbol.py

```diff
@@ -248,10 +248,18 @@ def resolve_type(self):
                   examining the Container from which it is imported.
         :rtype: subclass of :py:class:`psyclone.psyir.symbols.Symbol`
 
+        :raises SymbolError: if the type could not be resolved.
         '''
         if self.is_import:
             extern_symbol = self.get_external_symbol()
+            # pylint: disable=import-outside-toplevel
             from psyclone.psyir.symbols import RoutineSymbol
+            # pylint: disable=unidiomatic-typecheck
+            if type(extern_symbol) is Symbol:
+                raise SymbolError(
+                    f"The external symbol '{extern_symbol.name}' was found "
+                    f"but it does not have a type. Maybe it is a transitive "
+                    f"import which is currently not resolvable.")
```

Comment:
Please could you add "...transitive (indirect) import which..." as, until you started using the term "transitive", I'd never heard of it.

Reply:
Done